druid/src/widget/split.rs

```diff
@@ -0,0 +1,327 @@
+    /// Create a new split panel which splits the horizontal area,
+    /// leaving the full height available to both children.
+    ///
+    /// Note that `split_point` must be between 0 and 1  
+    pub fn new<T: Data>(
+        split_point: f64,
+        splitter_size: f64,
+        draggable: bool,
+        child1: impl Widget<T> + 'static,
+        child2: impl Widget<T> + 'static,
+    ) -> Split<T> {
+        Split::new(
+            Axis::Horizontal,
+            split_point,
+            splitter_size,
+            draggable,
+            child1,
+            child2,
+        )
+    }
+}
+/// A builder for a widget which divides space vertically
+/// The splitter bar will run left-to-right
+///
+/// The actual widget it implemented by [`Split`], but this builder
+/// is provided for convenience.
+///
+/// [`Split`]: struct.Split.html
+pub struct VerticalSplit;
+impl VerticalSplit {
+    /// Create a new split panel which splits the vertical area,
+    /// leaving the full width available to both children.
+    ///
+    /// Note that `split_point` must be between 0 and 1  
+    pub fn new<T: Data>(
+        split_point: f64,
+        splitter_size: f64,
+        draggable: bool,
+        child1: impl Widget<T> + 'static,
+        child2: impl Widget<T> + 'static,
+    ) -> Split<T> {
+        Split::new(
+            Axis::Vertical,
+            split_point,
+            splitter_size,
+            draggable,
+            child1,
+            child2,
+        )
+    }
+}
+
+impl<T: Data> Split<T> {
+    ///Create a new split panel.
+    ///Note that split_point must be between 0 and 1.
+    fn new(
+        split_direction: Axis,
+        split_point: f64,
+        splitter_size: f64,
+        draggable: bool,
+        child1: impl Widget<T> + 'static,
+        child2: impl Widget<T> + 'static,
+    ) -> Split<T> {
+        assert!(
+            split_point > 0.0 && split_point < 1.0,
+            "Invalid split point {} provided!",
+            split_point
+        );
+        Split {
+            split_point,
+            split_direction,
+            splitter_size,
+            draggable,
+            child1: WidgetPod::new(child1).boxed(),
+            child2: WidgetPod::new(child2).boxed(),
+        }
+    }
+    fn splitter_hit_test(&self, size: Size, mouse_pos: Point) -> bool {
+        match self.split_direction {
+            Axis::Horizontal => {
+                let center = size.width * self.split_point;
+                (center - mouse_pos.x).abs() < self.splitter_size / 2.0
+            }
+            Axis::Vertical => {
+                let center = size.height * self.split_point;
+                (center - mouse_pos.y).abs() < self.splitter_size / 2.0
+            }
+        }
+    }
+    fn update_splitter(&mut self, size: Size, mouse_pos: Point) {
+        self.split_point = match self.split_direction {
+            Axis::Horizontal => {
+                let max_limit = size.width - (self.splitter_size * 0.5);
```

Comment:
If I make the splitter large enough I can end up with constraints that are negative sizes, which is undesirable. Not sure the best approach, but we should be limiting this somehow.

Reply:
I may clamp-to-0 this out, but I suspect I need to do a little better check in my extrema-limiting in `adjust_splitter`. Probably some floating-point error allowing that to happen. I can probably mitigate it with some more 'fuzz', similar to what we might do in the hit test

Reply:
I went with more of a fuzz approach by attaching a minimum size 'gap' that can be provided by doing `(splitter_size*0.5).min(5.0)`. 

Gotta run for now, but will test this tomorrow to see if I need to increase/decrease that number. Otherwise, I'll adjust `layout` to clamp the numbers provided to the BoxConstraints to 0 when propagating them down (not that I like that option - it feels like a kludge for a failure in this algorithm).